library/alloc/src/vec.rs

```diff
@@ -2928,6 +2925,23 @@ impl<T> IntoIter<T> {
         self.ptr = self.buf.as_ptr();
         self.end = self.buf.as_ptr();
     }
+
+    /// Shifts the remaining elements to the front and then converts the whole allocation to a Vec
+    fn into_vec(self) -> Vec<T> {
+        if self.ptr != self.buf.as_ptr() as *const _ {
+            unsafe { ptr::copy(self.ptr, self.buf.as_ptr(), self.len()) }
+        }
+
+        let iter = ManuallyDrop::new(self);
+        unsafe { Vec::from_raw_parts(iter.buf.as_ptr(), iter.len(), iter.cap) }
+    }
+
+    fn move_to(mut self, dest: &mut Vec<T>) {
```

Comment:
I think it would be good to put some comments for this, I need to take a look to see what this does.
```suggestion
    /// Move remaining elements to the end of `dest`.
    fn move_to(mut self, dest: &mut Vec<T>) {
```